backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/workpackage/impl/AbstractMaterialReceiptLogHandler.java

```diff
@@ -84,38 +85,35 @@ public abstract class AbstractMaterialReceiptLogHandler implements IModularContr
 	public @NonNull ExplainedOptional<LogEntryCreateRequest> createLogEntryCreateRequest(
 			@NonNull final IModularContractLogHandler.CreateLogRequest request)
 	{
-		final TableRecordReference recordRef = request.getRecordRef();
-		final I_M_InOutLine inOutLineRecord = inOutBL.getLineByIdInTrx(InOutLineId.ofRepoId(recordRef.getRecordIdAssumingTableName(getSupportedTableName())));
-		final I_M_InOut inOutRecord = inOutBL.getById(InOutId.ofRepoId(inOutLineRecord.getM_InOut_ID()));
-		final I_C_Flatrate_Term flatrateTermRecord = flatrateDAO.getById(request.getContractId());
-		final Quantity quantity = inOutBL.getQtyEntered(inOutLineRecord);
+		final InOutLineId receiptLineId = getReceiptLineId(request.getRecordRef());
+		final I_M_InOutLine receiptLineRecord = inOutBL.getLineByIdInTrx(receiptLineId);
+		final I_M_InOut receiptRecord = inOutBL.getById(InOutId.ofRepoId(receiptLineRecord.getM_InOut_ID()));
+		final I_C_Flatrate_Term contractRecord = flatrateDAO.getById(request.getContractId());
+		final Quantity quantity = inOutBL.getQtyEntered(receiptLineRecord);
 
-		final ProductId productId = ProductId.ofRepoId(inOutLineRecord.getM_Product_ID());
+		final ProductId productId = request.getProductId();
 		final String productName = productBL.getProductValueAndName(productId);
 		final String description = msgBL.getBaseLanguageMsg(MSG_ON_COMPLETE_DESCRIPTION, productName, quantity);
 
 		final boolean isBillable = getLogEntryContractType().isInterimContractType()
-				? inOutRecord.isInterimInvoiceable()
+				? receiptRecord.isInterimInvoiceable()
 				: true;
 
-		final LocalDateAndOrgId transactionDate = LocalDateAndOrgId.ofTimestamp(inOutRecord.getMovementDate(),
-				OrgId.ofRepoId(inOutLineRecord.getAD_Org_ID()),
-				orgDAO::getTimeZone);
+		final LocalDateAndOrgId transactionDate = extractMovementDate(receiptRecord);
 
 		final InvoicingGroupId invoicingGroupId = modCntrInvoicingGroupRepository.getInvoicingGroupIdFor(productId, transactionDate.toInstant(orgDAO::getTimeZone))
 				.orElse(null);
 
-		final ProductPrice contractSpecificPrice = modularContractService.getContractSpecificPrice(request.getModularContractModuleId(),
-				request.getContractId());
+		final ProductPrice contractSpecificPrice = getPriceActual(request);
 
 		return ExplainedOptional.of(LogEntryCreateRequest.builder()
 				.contractId(request.getContractId())
-				.productId(ProductId.ofRepoId(inOutLineRecord.getM_Product_ID()))
-				.referencedRecord(TableRecordReference.of(I_M_InOutLine.Table_Name, inOutLineRecord.getM_InOutLine_ID()))
-				.collectionPointBPartnerId(BPartnerId.ofRepoId(inOutRecord.getC_BPartner_ID()))
-				.producerBPartnerId(BPartnerId.ofRepoId(inOutRecord.getC_BPartner_ID()))
-				.invoicingBPartnerId(BPartnerId.ofRepoId(flatrateTermRecord.getBill_BPartner_ID()))
-				.warehouseId(WarehouseId.ofRepoId(inOutRecord.getM_Warehouse_ID()))
+				.productId(productId)
+				.referencedRecord(TableRecordReference.of(I_M_InOutLine.Table_Name, receiptLineId))
```

Comment:
`request.getRecordRef()` could be used or at least use `getSupportedTableName()`